.github/workflows/go-cd.yml

```
@@ -0,0 +1,186 @@
+
+  create-binaries:
+    needs: [load-platform-matrix, validate-release-version]
+    if: success() && github.repository_owner == 'valkey-io'
+    strategy:
+      fail-fast: false
+      matrix:
+        host: ${{ fromJson(needs.load-platform-matrix.outputs.PLATFORM_MATRIX)}}
+    runs-on: ${{ matrix.host.RUNNER}}
+    steps:
+      - name: Setup self-hosted runner access
+        run: |
+          GHA_HOME=/home/ubuntu/actions-runner/_work/valkey-glide
+          if [ -d $GHA_HOME ]; then
+            sudo chown -R $USER:$USER $GHA_HOME
+            sudo rm -rf $GHA_HOME
+            mkdir -p $GHA_HOME/valkey-glide
+          else
+            echo "No cleaning needed"
+          fi
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "^1.22.0"
+      - name: Install shared software dependencies
+        uses: ./.github/workflows/install-shared-dependencies
+        with:
+          os: ${{ matrix.host.OS }}
+          target: ${{ matrix.host.TARGET }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build Go client
+        working-directory: go
+        run: |
+          make install-build-tools
+          make build-glide-client
+          # TODO: move generation of protobuf and header file to a non-matrix step
+          make generate-protobuf
+      - name: Upload artifacts
+        continue-on-error: true
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.host.TARGET }}
+          path: |
+            go/target/release/libglide_rs.a
+            go/lib.h
+            go/protobuf/
+
+  commit-auto-gen-files-with-tag:
+    if: ${{ github.event.inputs.pkg_go_dev_publish }}
+    needs: [validate-release-version, create-binaries]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          path: artifacts
+      - name: Copy generated files to repo
+        run: |
+          cd artifacts
+          for dir in */; do
+            target_name=${dir%/}
+            mkdir -p $GITHUB_WORKSPACE/go/rustbin/${target_name}
+            cp ${target_name}/target/release/libglide_rs.a $GITHUB_WORKSPACE/go/rustbin/${target_name}/
+          done
+          # TODO: move generation of protobuf and header file to a non-matrix step
+          cd x86_64-unknown-linux-gnu
+          cp lib.h $GITHUB_WORKSPACE/go/
+          mkdir -p $GITHUB_WORKSPACE/go/protobuf
+          cp protobuf/* $GITHUB_WORKSPACE/go/protobuf/
+      - name: Commit and push tag
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add -f .
+          git commit -m "Automated commit from GitHub Actions"
+          git tag go/${{ needs.validate-release-version.outputs.RELEASE_VERSION}}
```

```suggestion
          git tag go/${{ needs.validate-release-version.outputs.RELEASE_VERSION }}
```
Update in all places (for readability)